### PR TITLE
feat: add criteria settings set based on task type

### PR DESCRIPTION
**Pull request by I-vasilich-I**
**🟢 Add `deploy` label if you want to deploy this Pull Request to staging environment**

#### 🧑‍⚖️ Pull Request Naming Convention

- Title should follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- Do not put issue id in title
- Do not put WIP in title. Use [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) functionality
- Consider to add `area:*` label(s)

* [x] I followed naming convention rules

---

#### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related issue link
[PR#2305](https://github.com/rolling-scopes/rsschool-app/pull/2305)
[issue#1980](https://github.com/rolling-scopes/rsschool-app/issues/1980)

#### 💡 Background and solution

- remove redundant `setState` call on form inputs changes
- add criteria settings set based on task type
admin-task
UI Update:
Screenshot from 2023-11-11 13-08-08

#### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Database migration is added or not needed
- [x] Documentation is updated/provided or not needed
- [x] Changes are tested locally

### CODE REVIEW

**Comment by I-vasilich-I** [comment]
Task types that I haven't put in any set, don't know if it should be in any and which
```js
/**
 * TaskDtoTypeEnum.Test
 * TaskDtoTypeEnum.Cvhtml
 * TaskDtoTypeEnum.Cvmarkdown
 */
 ```
 on [figma layout](https://www.figma.com/file/D31S4jMK4EQfnpvguVjc8W/RS-Redesign_Ant-design?node-id=133996%3A484955&t=9C3Ek8ek1zviEawi-1) mentioned two types for cross-check that doesn't exist in code base:
- Cross-check
- Group task

Need someone to double check settings set, and clarify what to do with types mentioned above

**Comment by apalchys** [comment]
* TaskDtoTypeEnum.Cvhtml
 * TaskDtoTypeEnum.Cvmarkdown
 
 this tasks need github if I'm not mistaken cc @valerydluski 


> on the issue and on [figma layout](https://www.figma.com/file/D31S4jMK4EQfnpvguVjc8W/RS-Redesign_Ant-design?node-id=133996%3A484955&t=9C3Ek8ek1zviEawi-1) mentioned two types for cross-check that doesn't exist in code base:

I think you can ignore

File: client/src/modules/Tasks/components/TaskModal/TaskModal.tsx [hunk 1]
```diff
@@ -47,36 +52,60 @@ export function TaskModal({
       ),
     [tasks],
   );
+  const taskTypes = useMemo(
```
Comment (apalchys) [inline client/src/modules/Tasks/components/TaskModal/TaskModal.tsx:55]: you don't need it inside component as it's static

**Review by apalchys — APPROVED** [review]

File: client/src/modules/Tasks/components/TaskModal/TaskModal.tsx (hunk 1 above)
Comment (I-vasilich-I) [reply client/src/modules/Tasks/components/TaskModal/TaskModal.tsx]: thanks, fixed.

**Review by valerydluski — APPROVED** [review]

**Pull request merged**